.github/workflows/test-portal.yml

```diff
@@ -0,0 +1,123 @@
+---
+name: Test Portal Experience
+
+##########################################
+# Start the job on push for all branches #
+##########################################
+
+# yamllint disable-line rule:truthy
+on:
+  pull_request:
+    types:
+      - edited
+      - opened
+      - reopened
+      - synchronize
+    paths:
+      - "eslzArm/**.json"
+      - "src/**.json"
+      - "src/**.bicep"
+  workflow_dispatch: {}
+
+env:
+  GITHUB_COMMIT_ID: ${{ github.event.pull_request.head.sha }}
+  GITHUB_PR_ID: ${{ github.event.pull_request.id }}
+  TEMP_SUBSCRIPTIONS_JSON_PATH: "./src/data/subscriptions.json"
+  TEMP_DEPLOYMENT_OBJECT_PATH: "./src/data/eslzArm.test.deployment.json"
+
+permissions:
+  contents: read
+  id-token: write
+
+###############
+# Set the Job #
+###############
+jobs:
+  test-portal:
+    name: Test Portal Experience
+    runs-on: ubuntu-latest
+    environment: csu-rw
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Show env
+        run: env | sort
+
+      - name: List available pwsh modules
+        uses: azure/powershell@v1
+        with:
+          inlineScript: Get-Module -ListAvailable
+          azPSVersion: "latest"
+
+      - name: Azure login (OIDC)
+        uses: azure/login@v1
+        if: ${{ success() && env.AZURE_CLIENT_SECRET == '' }}
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+          enable-AzPSSession: true
+        env:
+          AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+
+      - name: Azure login (Client Secret)
+        uses: azure/login@v1
+        if: ${{ success() && env.AZURE_CLIENT_SECRET != '' }}
+        with:
+          creds: '{"clientId":"${{ secrets.AZURE_CLIENT_ID }}","clientSecret":"${{ secrets.AZURE_CLIENT_SECRET }}","subscriptionId":"${{ secrets.AZURE_SUBSCRIPTION_ID }}","tenantId":"${{ secrets.AZURE_TENANT_ID }}"}'
+          enable-AzPSSession: true
+        env:
+          AZURE_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+
+      - name: Register subscriptions
+        uses: azure/powershell@v1
+        with:
+          inlineScript: |
+            ./src/scripts/Invoke-ActionRegisterSubscriptions.ps1
```

Comment:
Do we not run the risk of collisions with this approach, i.e. if more PRs are approved independently? Would it make more sense to generate new subscriptions for each deployment and trash after PR close?

Reply:
Good question. This is a matter of choice ultimately and I prefer the idea that we co-ordinate runs for PRs, keeping in mind the only "collision" per-say is the placement of Subscriptions in a Management Group. All other resources (and the corresponding clean-up activities) are run on the basis of the unique `rootId` provided under each PR so we could theoretically run multiple PRs in parallel without too many issues.

However I don't consider this to be a likely event given current rate of change, and IMHO that downside far outweighs the issues associated with a high churn rate of Subscription creation and deletion operations needed to ensure unique Subscriptions per PR.